### PR TITLE
Add sonar cloud pipeline

### PR DESCRIPTION
**Pull request by PlesnikJakub**
Hi guys 👋 
I am testing sonar cloud features and I think it could be a great addition to this project.
It's free for public repositories. ❤️ 

Let me know if you have any comments. 😊

### CODE REVIEW

**Comment by petrspelos** [comment]
Great work, @PlesnikJakub! Looks awesome. ❤️ 

Though it seems to be confused by Blazor at the moment, not seeing its base methods for overrides. 
Is there a way to perhaps ignore certain suggestions? 🤔

**Comment by PlesnikJakub** [comment]
> Though it seems to be confused by Blazor at the moment, not seeing its base methods for overrides.
> Is there a way to perhaps ignore certain suggestions? 🤔

To be honest I am not really sure 🤔
I was waiting to see what will be the result in the web app (there will be result only for the main branch).

**Comment by PlesnikJakub** [comment]
@fpindej @thezerda any opinions guys? 🤔

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          java-version: 1.11
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0'
+          include-prerelease: True
+      - name: Setup .NET
```
Comment (fpindej) [inline .github/workflows/main.yml:25]: Are you setting up both .NET 6 and .NET 5 on purpose? 🤔

**Comment by fpindej** [comment]
@PlesnikJakub Static code analysis tool is a great addition to project, we used something similar in the past. As discussed with You before: this is a great project to try and set it up for future use on other projects as well.

I also noticed you're running `Setup .NET` twice, is that on purpose?

**Comment by fpindej** [comment]
How much control do we have about what we consider a code smell, vulnerability etc.? 🤔

**Comment by PlesnikJakub** [comment]
> @PlesnikJakub Static code analysis tool is a great addition to project, we used something similar in the past. As discussed with You before: this is a great project to try and set it up for future use on other projects as well.
> 
> I also noticed you're running `Setup .NET` twice, is that on purpose?

yes, it's on purpose, the project requires .NET 6 but for some reason, SonarClour requires also .NET 5..

**Comment by PlesnikJakub** [comment]
> How much control do we have about what we consider a code smell, vulnerability, etc.? 🤔

On this front I am also not sure, I would say, let's merge it to main, see the results of the analysis and then we can decide more. 
Without that, I am only guessing from what I saw on previous projects.

File: .github/workflows/main.yml (hunk 1 above)
Comment (PlesnikJakub) [reply .github/workflows/main.yml:25]: YES. without both of them, the Project or SonarCloud tool won't build... It's weird but you can check previous runs of this pipeline without .NET 5 👇 

https://github.com/sde-sw/blazor-interview-app/runs/3523331996?check_suite_focus=true

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
```
Comment (rzerdik) [inline .github/workflows/main.yml]: Looks great! Thank you very much for this. 
I was only wondering if this `master` branch is correct as we have `main` branch?

Comment (fpindej) [reply .github/workflows/main.yml]: You're right! Good catch 🧤

Comment (fpindej) [reply .github/workflows/main.yml]:
```suggestion
      - main
```

**Review by fpindej — CHANGES_REQUESTED** [review]

**Comment by fpindej** [comment]
@PlesnikJakub commit the suggestion and we'll merge it to `main` 😊

**Review by fpindej — APPROVED** [review]

**Pull request merged**